### PR TITLE
fix: NXDataStore cannot store numbers

### LINKED ISSUE #5722: EFB PTU Sound setting not saved
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-08-25T07:03:31+00:00",
  "ref": "master",
  "sha": "eb7ab6f30c5e276062cde6cb23b6e261afbee09f",
  "actor": "Benjozork",
  "event_name": "manual"
}
```

### Describe the bug

Recently I found that I did not hear the PTU sound anymore. When checking the EFB this setting was disabled. I enabled it and could hear the PTU sound. However, after closing and starting the sim again the setting is again defaulted to off...

### Expected behavior

I would expect the setting is stored so you only need to touch it when you want to change it.

### Steps to reproduce

1. Start a flight, check the EFB for the PTU sound setting
2. Verify it is OFF
3. Set it to ON and do a flight as normal
4. Check the PTU sound on start-up
5. Close MSFS
6. Start MSFS again and check the EFB setting again...

### References (optional)

PTUSetting

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### LINKED ISSUE #5751: EFB Brightness Option Not Saving
### Aircraft Version

Experimental

### Build info

```json
{
  "built": "2021-08-26T10:35:24+00:00",
  "ref": "experimental",
  "sha": "14a5718876e99aefde0a96682f609f7dc2e82b99",
  "actor": "aguther",
  "event_name": "manual"
}
```

### Describe the bug

Brightness Option in the EFB is not saving after reload. This probably is happening because I used auto brightness and decided to turn it back off. (Please check references and/or steps to reproduce)

### Expected behavior

EFB to remember last Brightness position.

### Steps to reproduce

I believe this could reproduce if you..

1. On the EFB navigate to Settings > flyPad
2. Turn on Auto Brightness
3. Set to night (optional: to clearly see bug)
4. Turn Auto Brightness back off while Time of Day is still night
5. Change Brightness to be brighter
6. Press ESC and exit to the MAIN MENU (do not RESTART)
7. Spawn back in and navigate back to EFB > Settings > flyPad > Brightness

### References (optional)

https://user-images.githubusercontent.com/72807945/131265685-ddac652b-49b7-40a1-815d-77fbc7553cb4.mp4

### Additional info (optional)

I'm running into a bunch of auto-saving issues with the EFB and just a thought if possibly could just have a "save" button for the settings. At least if I click a "save" button then I know 100% my stuff was saved.

### Discord Username (optional)

Aviation Spotter#9102

### PR DESCRIPTION
**Pull request by tracernz**
The underlying Coherent datastore errors out when anything other than a
string is set.

Fixes #5722
Fixes  #5751

## Summary of Changes

The Coherent datastore cannot store anything other than strings. Our typescript NXDataStore implementation was attempting to store numbers. The exception was being caught by the SetStoredData function in JS, which returned null, and we ignored that return value so the failure was silent.

Ammed the NXDataStore class to only take strings as it's type and adjust consumers to use strings. Hopefully we can come up with a solution later, but TS overloads don't seem to be useful here, and generics won't do it.

Additionally the SimVarSyncedProperty did not make a lot of sense, as it was attempting to sync in the opposite direction to what _all_ of the consumers desired. Use the new NXDataStore syncing functionality to sync settings across instruments instead, and make the one way sync from the setting to the simvar, not the other way around.

Tagging for 0.7 at this stage, as the bug would be quite annoying to release.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Make sure sound and EFB brightness settings are persisted across MSFS restarts (not Alt+F4 ones!).

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/shared/src/persistence.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export type StorageValue = string | number;
+export type StorageValue = string; // this is the ONLY type that SetStoredData accepts, it silently fails otherwise
```
Comment (Benjozork) [inline src/shared/src/persistence.ts:1]: Should we not just remove the type ?

Comment (tracernz) [reply src/shared/src/persistence.ts:1]: I didn't want to do that just yet... although we probably will have to unless we can find a trick to allow us to make it work.

Comment (Benjozork) [reply src/shared/src/persistence.ts:1]: > although we probably will have to unless we can find a trick to allow us to make it work.

What do you mean by that exactly ?

Comment (tracernz) [reply src/shared/src/persistence.ts]: Resolved on Discord. I have added a usePersistentNumberProperty hook for users that want to store numbers.

**Review by Benjozork — APPROVED** [review]

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Tester Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #PR 5741
Tier of Testing: 1
Date: 29/08/2021

Testing Process:
- Loaded in on runway and C+D at the gate at EGSS over multiple full restarts.
- Changed sound and EFB brightness settings for persistence - PASS

Note: Restarted by exiting to main menu, then exiting to desktop, not ALT+F4.

Negatives: N/A
Testing Results: PASS 

Conclusions: Fixes persistence with EFB settings, no other issues observed, great stuff!

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #5741
Tier of Testing: 1
Date: 30.08.2021

Testing Process: 
C&D, Change brightness and sound settings.
Shutdown sim.
Startup again and see if settings were stored. 
Repeated this twice

Negatives: none 
Testing Results: Passed
Conclusions: LGTM

**Comment by frankkopp** [comment]
Addendum to my test:

Thrust Reduction Height, Acceleration Height, Engine Out values were cleared at the first attempt. 
After setting them they were persisted from then on. 

I'm not sure if it causes an issue if the fields are empty. If so this must be fixed before merge. 

image

**Comment by FlightSimmer7700** [comment]
Quality Assurance Tester

Discord: FlightSimmer7700#3468
Object of testing: #5741 
Tier of Testing: 1
Date: 29/8/21

Testing Process: 
Loaded in and changed settings for brightness and sound
Closed the sim 
Reopened the sim
Checked all settings were correct 

Negatives: When setting autobrightness after having set manual brightness the slider does not move to the brightness selected by the autobrightness but instead stays where it was. Then when autobrightness is turned off the slider still stays where it was but the brightness of the efb stays on what was last set by the 
Testing Results: Not Passed

Conclusions: Sound settings save as expected but the brightness setting slider is a little buggy compared to the version in dev
Media:
https://streamable.com/r5rvlt

**Comment by NUTTTTTTTT** [comment]
> Quality Assurance Tester
> 
> Discord: FlightSimmer7700#3468
> Object of testing: #5741
> Tier of Testing: 1
> Date: 29/8/21
> 
> Testing Process:
> Loaded in and changed settings for brightness and sound
> Closed the sim
> Reopened the sim
> Checked all settings were correct
> 
> Negatives: When setting autobrightness after having set manual brightness the slider does not move to the brightness selected by the autobrightness but instead stays where it was. Then when autobrightness is turned off the slider still stays where it was but the brightness of the efb stays on what was last set by the
> Testing Results: Not Passed
> 
> Conclusions: Sound settings save as expected but the brightness setting slider is a little buggy compared to the version in dev
> Media:
> https://streamable.com/r5rvlt

Tested following the above process, able to verify behavior - amended report based on this

**Comment by thesimpilot** [comment]
Quality Assurance Tester/Trainee Report

Object of testing: #5741
Tier of Testing: 1
Date: 08/29/2021

Testing Process: Loaded C&D at EDDM, turned on EFB while C&D set the Audio and Brightness levels and turned on Auto Brightness. Closed the sim, re-launched and loaded at EDDM different gate than first load. Turned on EFB while C&D checked previously set levels to confirm persistence. Audio levels were persistent and turning off auto brightness restored the previous setting set initially. Left Auto Brightness off, closed the sim re-launched and loaded to a different airport (EDDF) the values for audio and brightness were observed as persistent with the initial launch.

Negatives: N/A

Testing Results: Passed

Conclusions: Brightness and audio levels are carried across the board after each sim restart.
Media: N/A

**Comment by FlightSimmer7700** [comment]
Quality Assurance Tester

Discord: FlightSimmer7700#3468
Object of testing: #5741 
Tier of Testing: 1
Date: 30/8/21

Testing Process: Loaded in to check brightness settings

Negatives: N/A
Testing Results: Passed

Conclusions: All lgtm

**Review by Benjozork — APPROVED** [review]

**Pull request merged**